### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Rishi05051997**
Peer Members, please review the Authentication Module, it contains below feature :
Login,
Signup, 
Change Password Screen

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,6 +8,14 @@ import "./styles/main.css";
 export default function App() {
   const { state: { toastMsg }, dispatch, isError } = useData();
   const [loader, setLoader] = useState(false);
+  const { login, setShowLoader } = useAuthContext();
```
Comment (maitrakhatri) [inline src/App.js:11]: `useAuth` is enough need not to add Context after it

File: src/components/Auth/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            const { createdUser } = await registerUser(firstName, lastName, email, password)
+            if (createdUser) {
+                ///// show success message
+                const { firstName } = createdUser;
+                dispatch({ type: "SHOW_TOAST", payload: `${firstName} created successfully in DB` })
+                navigate("/")
+            }
+        } catch (error) {
+            // setErrorMsg(error);
+            // setShowMsg(true)
```
Comment (maitrakhatri) [inline src/components/Auth/Signup.jsx]: remove comments if not needed

File: src/context/auth-context.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+    const loginUser = async (email, password) => {
+        if (!isPasswordValid(password)) {
+            setErrorMsg("Password must contain at least 8 characters, at least 1 number and both lower and uppercase letters.")
+            setShowMsg(true);
+        } else {
+            try {
+                const { data: { foundUser, encodedToken } } = await axios({
+                    method: 'post',
+                    url: `${API_URL}/auth/login`,
```
Comment (maitrakhatri) [inline src/context/auth-context.js]: this can be reduced to `axios.get('${API_URL}/auth/login', {data})`

File: src/context/auth-context.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                    data: {
+                        email,
+                        password
+                    }
+                })
+                localStorage.setItem("token", encodedToken)
+                // console.log(user)
+                // const decoded = jwt_decode(token);
+                // const loginData = { token: `Bearer ${token}`, user: decoded.name };
```
Comment (maitrakhatri) [inline src/context/auth-context.js]: remove console logs if not needed

File: src/context/auth-context.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        }
+        else if (!isPasswordValid(password)) {
+            setErrorMsg("Password must contain at least 8 characters, at least 1 number and both lower and uppercase letters.")
+            setShowMsg(true);
+        } else {
+            try {
+                const { data } = await axios({
+                    method: "post",
+                    url: `${API_URL}/auth/signup`,
+                    data: { firstName, lastName, email, password }
```
Comment (maitrakhatri) [inline src/context/auth-context.js]: same as above, this can be reduced to one line

**Review by maitrakhatri — APPROVED** [review]
Clean code, easy to read and understand.
Left some suggestions

**Review by poornima-alugubelly — APPROVED** [review]
Clean code everything seems good to me

**Pull request merged**